cmake/SettingsRegistry.cmake

```
@@ -54,6 +61,21 @@ function(ly_get_gem_load_dependencies ly_GEM_LOAD_DEPENDENCIES ly_TARGET)
         return()
     endif()
 
+    # detect cycles
+    if(ly_TARGET IN_LIST ly_CYCLE_DETECTION_TARGETS)
+        unset(dependency_cycle_loop)
+        foreach(item IN LISTS ly_CYCLE_DETECTION_TARGETS)
+            string(APPEND dependency_cycle_loop ${item})
+            if(item STREQUAL ly_TARGET)
+                string(APPEND dependency_cycle_loop "*")
+            endif()
+            string(APPEND dependency_cycle_loop " --> ")
+        endforeach()
+        string(APPEND dependency_cycle_loop " ${ly_TARGET}*")
+        message(FATAL_ERROR "Runtime dependency detected: ${dependency_cycle_loop}")
+    endif()
```

Can we move this to a `ly_detect_gem_cycle` function?
That way we can keep the `ly_gem_load_dependencies` function a little bit simpler
```suggestion
function(ly_detect_gem_cycle target visited_target_set)
    # detect cycles
    if(target IN_LIST visited_target_set)
        unset(dependency_cycle_loop)
        foreach(item IN LISTS visited_target_set)
            string(APPEND dependency_cycle_loop ${item})
            if(item STREQUAL target)
                string(APPEND dependency_cycle_loop "*")
            endif()
            string(APPEND dependency_cycle_loop " --> ")
        endforeach()
        string(APPEND dependency_cycle_loop " ${target}*")
        message(FATAL_ERROR "Runtime dependency detected: ${dependency_cycle_loop}")
    endif()
endfunction()
```

---

will do, I also will write it a bit more generic since it is just a loop detection through visitation algorithm